### PR TITLE
[CARBONDATA-2047] Clean up temp folder after task completion in case of partitioning

### PR DESCRIPTION
**Pull request by ravipesala**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2873/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1641/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1666/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/datasources/CarbonFileFormat.scala [hunk 1]
```diff
@@ -202,7 +231,7 @@ private trait AbstractCarbonOutputWriter {
 
 private class CarbonOutputWriter(path: String,
     context: TaskAttemptContext,
-    fieldTypes: Seq[DataType])
+    fieldTypes: Seq[DataType], taskNo : String)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/datasources/CarbonFileFormat.scala]: move taskNo to next line

File: processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java [hunk 2]
```diff
@@ -317,8 +313,7 @@ public void initializeWriter() throws CarbonDataWriterException {
         .getCarbonDataFileName(fileCount, model.getCarbonDataFileAttributes().getTaskId(),
             model.getBucketId(), model.getTaskExtension(),
             "" + model.getCarbonDataFileAttributes().getFactTimeStamp());
-    this.carbonDataFileTempPath = chosenTempLocation + File.separator
-        + carbonDataFileName + CarbonCommonConstants.FILE_INPROGRESS_STATUS;
+    this.carbonDataFileTempPath = chosenTempLocation + File.separator + carbonDataFileName;
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java]: For the temp location, can we add a "carbon" prefix to the path name, currently it is just a nanotime number, I think it is better if user can tell which folder is created by carbon, even it is temp folder

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2900/

File: processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java (hunk 2 above)
Comment (ravipesala) [reply processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java]: ok

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/datasources/CarbonFileFormat.scala (hunk 1 above)
Comment (ravipesala) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/datasources/CarbonFileFormat.scala]: ok

File: processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java (hunk 2 above)
Comment (ravipesala) [reply processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1687/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2920/

**Comment by jackylk** [comment]
It seems there are two testcase failure. And please change PR title

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1695/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2928/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2947/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1699/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2932/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2958/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2950/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1721/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2975/

**Pull request closed without merging**